src/tools/tidy/src/walk.rs

```diff
@@ -7,7 +7,6 @@ use ignore::DirEntry;
 
 /// The default directory filter.
 pub fn filter_dirs(path: &Path) -> bool {
-    // FIXME: sync submodule exclusion list with rustfmt.toml
```

Comment:
Discussion: @onur-ozkan should tidy be reading from rustfmt.toml or we somehow make sure that there's a single source of truth for this? (not for this PR)

Reply:
Skip lists for `rustfmt` and `tidy` seem not overlapping but if `tidy` needs to include the skip list of `rustfmt`, I wanna work on it.

Reply:
![image](https://github.com/user-attachments/assets/dedca148-ec0d-40c6-aed6-0bdc956fc8d2)

My other concern is that highlighted rows in the screenshot above (from `rustfmt.toml`) are not submodules. Maybe some time ago, they were and added to that list. But now they are managed inside the `rust` repo and `rustfmt` doesn't format them. I've tried to run `x fmt` for them and found a lot of errors. Shouldn't they be formatted like other files in the `rust` repo as well?

Reply:
Some of them are not submodules but subtrees and managed by `josh`, so I don't think this is easily sync'd.

Reply:
IIRC we don't want tidy to rely on a toml parser because it needs to build fast, so probably not sync'd easily.

Reply:
So how about relying on `.gitmodules` file for both rustfmt and tidy?